### PR TITLE
[Azure.Monitor.Ingestion] Fix flaky tests and rename EventArgs

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs [hunk 1]
```diff
@@ -392,33 +395,34 @@ public virtual async Task<Response> UploadAsync<T>(string ruleId, string streamN
...
+                if (ex is OperationCanceledException && cancellationToken.IsCancellationRequested)
                 {
-                    // We do not want to log exceptions here as we will loop through all the tasks later
-                    // Cancel all future Uploads if user triggers CancellationToken
-                    if (ex is OperationCanceledException && cancellationToken.IsCancellationRequested)
-                    {
-                        shouldAbort = true;
-                        AddException(ref exceptions, ex);
-                    }
+                    // do not add to Exception list here as we add when we process all runningTasks
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs:414]: This comment is out of date.

**Comment by nisha-bhatia** [comment]
\azp run net - Azure.Monitor.Ingestion - weekly

File: sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs [hunk 2]
```diff
@@ -442,18 +444,25 @@ public virtual async Task<Response> UploadAsync<T>(string ruleId, string streamN
             }
 
             // If no exceptions return response
-            return runningTasks.Select(_ => _.CurrentTask).Last().Result; //204 - response of last batch with header
+            if (runningTasks.Count == 0)
+                return firstTask.Result;
+
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs:449]:
```suggestion
```

File: sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs [hunk 3]
```diff
@@ -442,18 +444,25 @@ public virtual async Task<Response> UploadAsync<T>(string ruleId, string streamN
...
+            else
+                return runningTasks.Select(_ => _.CurrentTask).Last().Result; //204 - response of last batch with header
         }
 
         private static void ProcessCompletedTask((Task<Response> CurrentTask, List<object> Logs) runningTask, ref List<Exception> exceptions, ref int logsFailed)
         {
+            // Only want to process task if it not canceled, processing canceled task throws exception
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs:456]:
```suggestion
            // If the task was canceled, the OperationCanceledException was already observed and tracked in our exception list.
```

File: sdk/monitor/Azure.Monitor.Ingestion/tests/MonitorIngestionLiveTest.cs [hunk 4]
```diff
@@ -148,16 +149,19 @@ public async Task ValidInputFromArrayAsJsonWithMultiBatchWithGzip()
...
-
+            LogsIngestionClient.Compression = "gzip";
+            LogsIngestionClient.SingleUploadThreshold = Kb;
             // Make the request
             LogsUploadOptions options = new LogsUploadOptions();
-            options.MaxConcurrency = 10;
-            Response response = await client.UploadAsync(TestEnvironment.DCRImmutableId, TestEnvironment.StreamName, GenerateEntries(8, Recording.Now.DateTime), options).ConfigureAwait(false);
+            options.MaxConcurrency = 8;
+            Response response = await client.UploadAsync(TestEnvironment.DCRImmutableId, TestEnvironment.StreamName, GenerateEntries(8000, Recording.Now.DateTime), options).ConfigureAwait(false);
+            Assert.Greater(policy.MaxCount, 1);
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Ingestion/tests/MonitorIngestionLiveTest.cs:164]: Does this fail if you move the line back into the foreach loop?

Comment (nisha-bhatia) [reply sdk/monitor/Azure.Monitor.Ingestion/tests/MonitorIngestionLiveTest.cs]: It fails! Concurrency is always 1

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**